convert2rhel/subscription.py

```diff
@@ -120,28 +134,43 @@ def register_system():
                 attempt + 1,
                 MAX_NUM_OF_ATTEMPTS_TO_SUBSCRIBE,
             )
+
         loggerinst.info(
             "%sRegistering the system using subscription-manager ...",
             attempt_msg,
         )
 
         registration_cmd = RegistrationCommand.from_tool_opts(tool_opts)
-        output, ret_code = registration_cmd()
-        if ret_code == 0:
-            # Handling a signal interrupt that was previously handled by
-            # subscription-manager.
-            if "user interrupted process" in output.lower():
-                raise KeyboardInterrupt
-            return
 
-        loggerinst.info("System registration failed with return code = %s" % str(ret_code))
-        sleep(REGISTRATION_ATTEMPT_DELAYS[attempt])
-        attempt += 1
-    loggerinst.critical("Unable to register the system through subscription-manager.")
+        try:
+            registration_cmd()
+            loggerinst.info("System registration succeeded.")
+        except Exception as e:
+            loggerinst.info("System registration failed with error: %s" % str(e))
+            sleep(REGISTRATION_ATTEMPT_DELAYS[attempt])
+            attempt += 1
+            continue
+
+        break
+
+    else:  # While-else
+        # We made the maximum number of subscription-manager retries and still failed
+        loggerinst.critical("Unable to register the system through subscription-manager.")
+
+    return None
 
 
 class RegistrationCommand(object):
-    def __init__(self, activation_key=None, org=None, username=None, password=None, server_url=None):
+    def __init__(
```

Comment:
You removed `server_url`. Because of that you need to remove the line `self.server_url = server_url` from the `__init__()` body.

Also, some unit tests expect the parameter to be present, e.g. `test_calling_registration_command_with_serverurl`.

Reply:
Yeah, there were a lot of these left to be fixed.  I wanted to get the code up where   it could be looked at asap.  I think I've now fixed all of these places.